src/App.tsx

```diff
@@ -1,27 +1,47 @@
-import React, {useEffect} from 'react';
-import {setupStore, RootStoreContext, RootStore} from './models';
-import {RootNavigator} from './navigation';
+import React, {useEffect, useState} from 'react';
+import {ActivityIndicator, StyleSheet, Text, View} from 'react-native';
 import {GestureHandlerRootView} from 'react-native-gesture-handler';
-import {StyleSheet} from 'react-native';
+import {RootStore, RootStoreContext, setupStore} from './models';
+import {RootNavigator} from './navigation';
 
 function App(): JSX.Element {
+  const [settingUp, setSettingUp] = useState(true);
+
   useEffect(() => {
-    setupStore();
+    setupStore().then(() => setSettingUp(false));
   });
 
-  return (
-    <GestureHandlerRootView style={styles.gestureRootView}>
-      <RootStoreContext.Provider value={RootStore}>
-        <RootNavigator />
-      </RootStoreContext.Provider>
-    </GestureHandlerRootView>
-  );
+  const renderContent = () => {
+    if (settingUp) {
+      return (
+        <View style={styles.setupScreen}>
+          <ActivityIndicator size={'large'} color={'red'} />
+          <Text>Setting Up App!</Text>
```

Comment:
Here instead of showing this loader we should continue showing the bootsplash and remove the splash only after RootNavigator is loaded